### PR TITLE
One due by note

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
## Purpose / Description
_Describe the problem or feature and motivation_

## Fixes
This fix issue #5456 (by me). The first commit copies Anki behavior. As I reported, this behavior is bugged, so the second commit fix Anki bug the same way I just fixed it in anki's code. See https://github.com/dae/anki/pull/339

## Approach
As in Anki (and in my Anki's PR).
Edit: it's not a PR anymore, but it is dae/anki@c2a6c3e47a9d88d28d5dbdd129db97a28f044c49.

## How Has This Been Tested?

I tried to reproduce both bugs in ankidroid, and saw that with this PR, all dues are correct. #5456 did disappear and Anki's bug did not appear.

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

### CODE REVIEW

**Comment by Arthur-Milchior** [comment]
Update:
This PR is now up to date with anki. Because DAE did accept my PR. So now it's up to date with dae/anki@c2a6c3e47a9d88d28d5dbdd129db97a28f044c49

File: AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java [hunk 1]
```diff
@@ -707,19 +707,25 @@ public void _remNotes(long[] ids) {
         String snids = Utils.ids2str(nids);
         HashMap<Long, HashMap<Integer, Long>> have = new HashMap<>();
         HashMap<Long, Long> dids = new HashMap<>();
+        HashMap<Long, Long> dues = new HashMap<>();
         Cursor cur = null;
         try {
-            cur = mDb.getDatabase().query("select id, nid, ord, did, odid from cards where nid in " + snids, null);
+            cur = mDb.getDatabase().query("select id, nid, ord, did, due, odue, odid, type from cards where nid in " + snids, null);
             while (cur.moveToNext()) {
+				long id = cur.getLong(0);
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java]: Indentation

**Comment by Arthur-Milchior** [comment]
Corrected

File: AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java (hunk 1 above)
Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java]: Corrected

Comment (timrae) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java]: To be clear, I just commented on that one line to mark the PR as "changes requested". That doesn't mean that is the only line that needs to be corrected; I recommend you to use Android Studio's auto-format tool on every PR.

Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java]: I understood it this way. While I did only use emacs for this change, I did check every line of this PR before resubmitting

Comment (timrae) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java]: OK, I made the comment because there are still tabs introduced in `Collection.java`

Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java]: My bad. i only did check the last commit, and not the full PR.
Corrected

**Pull request merged**